packages/@react-aria/landmark/src/useLandmark.ts

```diff
@@ -0,0 +1,241 @@
+    }
+    
+    let currentLandmark = this.closestLandmark(element);
+    let nextLandmarkIndex = backward ? -1 : 0;
+    if (currentLandmark) {
+      nextLandmarkIndex = this.landmarks.findIndex(landmark => landmark === currentLandmark) + (backward ? -1 : 1);
+    }
+
+    // Wrap if necessary
+    if (nextLandmarkIndex < 0) {
+      nextLandmarkIndex = this.landmarks.length - 1;
+    } else if (nextLandmarkIndex >= this.landmarks.length) {
+      nextLandmarkIndex = 0;
+    }
+
+    return this.landmarks[nextLandmarkIndex];
+  }
+
+  /**
+   * Look at next landmark. If an element was previously focused inside, restore focus there.
+   * If not, focus the first focusable element inside the lanemark.
+   * If no focusable elements inside, go to the next landmark.
+   * If no landmarks at all, or none with focusable elements, don't move focus.
+   */
+  public f6Handler(e: KeyboardEvent) {
+    if (e.key === 'F6') {
+      e.preventDefault();
+      e.stopPropagation();
+
+      let backward = e.shiftKey;
+      let nextLandmark = this.getNextLandmark(e.target as HTMLElement, {backward});
+
+      // If no landmarks, return
+      if (!nextLandmark) {
+        return;
+      }
+
+      // If something was previously focused in the next landmark, then return focus to it
+      if (nextLandmark.lastFocused) {
+        let lastFocused = nextLandmark.lastFocused;
+        if (document.body.contains(lastFocused)) {
+          lastFocused.focus();
+          return;
+        }
+      }
+
+      // Otherwise, focus the landmark itself
+      if (document.contains(nextLandmark.ref.current)) {
+        nextLandmark.ref.current.focus();
+      }
+    }
+  }
+
+  /**
+   * Sets lastFocused for a landmark, if focus is moved within that landmark.
+   */
+  public focusinHandler(e: FocusEvent) {
+    let currentLandmark = this.closestLandmark(e.target as HTMLElement);
+    if (currentLandmark) {
+      this.updateLandmark({...currentLandmark, lastFocused: e.target as HTMLElement});
+    }
+  }
+}
+
+/**
+ * Provides landmark navigation in an application. Call this with a role and label to register a landmark navigable with F6.
+ * @param props - Props for the landmark.
+ * @param ref - Ref to the landmark.
+ */
+export function useLandmark(props: AriaLandmarkProps, ref: MutableRefObject<HTMLElement>): LandmarkAria {
+  const {
+    role,
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledby
+  } = props;
+  let manager = LandmarkManager.getInstance();
+  let label = ariaLabel || ariaLabelledby;
+
+  useLayoutEffect(() => {
+    manager.addLandmark({ref, role, label});
+    
+    return () => {
+      manager.removeLandmark(ref);
+    };
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    manager.updateLandmark({ref, label, role});
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [label, ref, role]);
+
+  return {
+    landmarkProps: {
+      role,
+      tabIndex: -1
```

Comment:
I think setting `tabIndex={-1}` on landmarks by default will become problematic, because it means that the landmark will always be focusable using the mouse. For example clicking on a static paragraph within the landmark will set focus to the landmark, which I'm not sure we want. We should only set `tabIndex={-1}` to a landmark just before we need it to receive programmatic focus, after a navigation event like pressing <key>F6</key>. We should then remove `tabIndex={-1}` when the landmark loses focus.

Reply:
Just pushed these changes, thanks!